### DISCUSSION TITLE
pvc clone with deep-flatten features.

### DISCUSSION THREAD

**Discussion by qkfrksvl** [ceph/ceph-csi#4465] [Q&A]
https://github.com/ceph/ceph-csi/blob/devel/docs/design/proposals/rbd-snap-clone.md#volume-cloning-datasource-pvc

I've read the documentation related to pvc clone and I have a few questions.

- Why use the clone function? I understand that this function has lower read/write performance than the CP function.
- When pvc clone, is deep-flatten applied automatically or can the cp function be used? (Currently performing deep-flatten manually)

**Comment by Rakshith-R** [comment]
@qkfrksvl That's the current design.

We maintain chain of rbd images and flatten at particular limits to balance ceph performance and to make full use of image layering. 

Can you point cp function documentation that you are referring to ?
